### ISSUE TITLE
Test failure on master

### ISSUE THREAD

**Issue by abhagupta** [swagger-api/swagger-js#945]
One of the test is failing on master branch.
```
 359 passing (5s)
  1 failing

  1) swagger resolver fails gracefully with bad spec:
     TypeError: undefined
```

Just making sure that this is expected. If not, is there any other branch or release branch public should be using?

**Comment by fehguy** [comment]
Build passes cleanly:

https://travis-ci.org/swagger-api/swagger-js

So not sure if you have a local environment issue or not.

**Comment by abhagupta** [comment]
Thanks for confirming that it is an expected failure. 
![image](https://cloud.githubusercontent.com/assets/825965/22664539/fdb909dc-ec65-11e6-8c26-5b2dc3522869.png)

I see the same thing.

**Comment by abhagupta** [comment]
@fehguy  if you run the tests on latest node, it does throw the following error:

```
1) swagger resolver fails gracefully with bad spec: 
TypeError: undefined
 at urlFormat (url.js:532:11)
 at new Request (node_modules/superagent/lib/node/index.js:136:37)
 at module.exports (node_modules/superagent/lib/node/index.js:41:10) 
 at Function.request.(anonymous function) [as get]  (node_modules/superagent/lib/node/index.js:986:15) 
 at SuperagentHttpClient.execute (lib/http.js:229:26)
 at SwaggerHttp.execute (lib/http.js:120:12) 
 at SwaggerClient.build (lib/client.js:275:23)
 at SwaggerClient.initialize (lib/client.js:185:17)
 at new module.exports (lib/client.js:109:17) at Context.<anonymous> (test/durability.js:14:5)
```

**Comment by fehguy** [comment]
@abhagupta which latest node?  Please give a version.

**Comment by abhagupta** [comment]
I am on 7.9.1.

**Comment by fehguy** [comment]
`7.9.1` isn't even available in nvm so I'm testing with the latest available there which is `v7.7.1`.

When I switch to that version, I indeed get the error you're seeing.

**Comment by abhagupta** [comment]
Oops. Sorry I meant 6.9.1
